src/classes/route.ts

```diff
@@ -0,0 +1,31 @@
+import { Restaurant, Location } from './location';
+import { Order } from './order'
+
+
+class Route {
+  private destinations: Location[] = [];
+  private orders: Order[] = [];
+
+  constructor(destinations: Location[] = [], orders: Order[] = []) {
+	  this.destinations = destinations;
+	  this.orders = orders;
+  }
+
+  getDestinations(): Location[] {
+	  return this.destinations;
+  }
+
+  getOrders(): Order[] {
+	  return this.orders;
+  }
+
+
+  next(): Location {
+	return this.destinations[0];
```

Comment:
¿Lo va a guardar borrando del array?

Reply:
No entiendo

Reply:
next siempre da el primero del array. El tema es, otra vez, el estado. ¿Cuando lleges a una "destination" vas a borrarla del array? Es esencial que guardéis y gestionéis el estado de un objeto de forma práctica.

Reply:
Cuando se haga todo lo que hay que hacer en la siguiente destination entonces la siquiente destination se actualiza automaticamente. Next siempre devuelve el proximo destino en el que ademas es seguro que queda algo por hacer.